src/main/resources/lib/task.ts

```diff
@@ -0,0 +1,43 @@
+import { Content, QueryResponse } from 'enonic-types/lib/content'
+import { Dataquery } from '../site/content-types/dataquery/dataquery'
+import { splitEvery } from 'ramda'
+import {withSuperUserContext} from "./repo/common";
+import {EVENT_LOG_BRANCH, EVENT_LOG_REPO} from "./repo/eventLog";
+import {User} from "enonic-types/lib/auth";
+const {
+  refreshDataset
+} = __non_webpack_require__('/lib/dataquery')
+const { run } = __non_webpack_require__('/lib/xp/context')
+
+const {
+  progress,
+  submit
+} = __non_webpack_require__('/lib/xp/task')
+
+const user: object = {
+  login: 'su',
+  userStore: 'system'
+}
+
+const master: object = { // Master context (XP)
+  repository: 'com.enonic.cms.default',
+  branch: 'master',
+  principals: ['role:system.admin'],
+  user
+}
+
+export function asynchronusRefreshRows(httpQueries: QueryResponse<Dataquery>, numberOfAsyncronusJobs: number = 4): Array<string> {
```

Comment:
_nitpick_: typo in name. the convention is also to simply call it `refreshRowsAsync` i think.. 

_naming_: 
1. litt usikker på om refreshRowsAsync kan brukes for andre type queries og, men likevel, er det litt bedre vi kaller den `refreshQueriesAsync`. 
1. `numberOfAsyncronusJobs` => `batchSize` or `chunkSize`
1. `chunksOfRows` => `httpQueriesBatch` or `httpQueriesChunk`
1. `rows` => `httpQueries`
1. `row` => `httpQuery`